### ISSUE TITLE
TSServer doesn't seem to pick up changes to the lib option

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#11398]
Using

* VS Code 1.5.3
* TS Version 2.1.0-dev.20161002

Start out with the following `tsconfig.json`:

```json
{
    "compilerOptions": {
        "module": "commonjs",
        "target": "es5",
        "noImplicitAny": true,
        "sourceMap": false,
        "lib": [
             "es5"
        ]
    }
}
```

and `./src/index.ts`:

```ts
declare var a: Promise<number>;
```

You should get the error `Cannot find name 'Promise'.`

Now add `es2015.promise` to `tsconfig.json`:

```diff
 {
     "compilerOptions": {
         "module": "commonjs",
         "target": "es5",
         "noImplicitAny": true,
         "sourceMap": false,
         "lib": [
+            "es2015.promise",
             "es5"
         ]
     }
 }
```

Expected: No error.
Actual: Error remains.
